marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/provenance/migrate-provenance.xqy

```diff
@@ -0,0 +1,236 @@
+xquery version "1.0-ml";
+
+module namespace migrate-prov = "http://marklogic.com/datahub/migrate-prov";
+
+declare namespace prov = "http://www.w3.org/ns/prov#";
+
+import module namespace config = "http://marklogic.com/data-hub/config" at "/com.marklogic.hub/config.xqy";
+import module namespace dh-prov = "http://marklogic.com/data-hub/dh-provenance-services" at "dh-provenance.xqy";
+
+declare private variable $collection as xs:string := "http://marklogic.com/provenance-services/record";
+declare private variable $user-role as xs:string := "ps-user";
+declare private variable $internal-role as xs:string := "ps-internal";
+
+declare function migrate-provenance($migrate-request as json:object, $endpoint-state as json:object) {
+  xdmp:security-assert("http://marklogic.com/data-hub/privileges/delete-provenance", "execute"),
+  let $batch-size := fn:head((map:get($migrate-request, "batchSize"), 250))
+  let $transform-results := xdmp:invoke-function(function() {
+    let $collection-query := cts:collection-query('http://marklogic.com/provenance-services/record')
+    let $last-migrated-uri-query := if (map:contains($endpoint-state, "lastUri")) then
```

Comment:
Do the "legacy" provenance documents get deleted? MigrateProvenanceTest doesn't make that clear, but I'm guessing they don't - and more generally, they're not updated in any fashion - and that's why we need the "last URI" approach here?

Reply:
Correct, they aren't touched. There is a recommendation after running the task to run the hubDeleteProvenance task against the JOBS DB once the user verifies the migration.